backends/dpdk/dpdkAsmOpt.h

```diff
@@ -135,6 +135,29 @@ class RemoveLabelAfterLabel : public Transform {
     }
 };
 
+
+// This pass Collects all metadata struct member used in program
+class CollectUsedMetadataField : public Inspector {
+    ordered_set<cstring> used_fields;
+  public:
+    bool preorder(const IR::Member *m) override {
+        if (m->toString().substr(0, 2) == "m.") {
```

Comment:
You could use startsWith() 
https://github.com/p4lang/p4c/blob/main/lib/cstring.h#L206